### PR TITLE
simulator: frequency management

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #2079

## What is the purpose of the change

This PR improves upon the previous `Weights` provided to specific actions. We instead want to move away from numerically defining how often actions should occur across modules and instead categorize them with human readable definitions (i.e. "Rare", "Common", "Frequent")

## Brief Changelog

- [x] Add an `WithFrequency(w Frequency) Action` method to the action interface
- [x] Change the type of Frequency to `string`
- [x] Make a frequency from int function, that we can use for legacy operations.
- [x] Move the action selection out of executor into types, and make it convert the frequencies into ints / replicate old functionality for now.
- [x] Add a demo API to some action:
  - `simtypes.NewMsgBasedAction("CreateUniV2Pool", am.keeper, gamm.CreateUniV2Pool).WithWeight(simtypes.Rare)`
- [x] Rename Weight to Freq

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: simulation/simtypes/weight.go [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+		return 5
+	case strFrequency == Infrequent:
+		return 15
+	case strFrequency == Common:
+		return 35
+	case strFrequency == Frequent:
+		return 65
+	default:
+		return 35
+	}
```
Comment (p0mvn) [inline simulation/simtypes/weight.go:39]:
```suggestion
	switch strFrequency {
	case Rare:
		return 5
	case Infrequent:
		return 15
	case Common:
		return 35
	case Frequent:
		return 65
	default:
		return 35
	}
```

File: simulation/simtypes/action.go [hunk 2]
```diff
@@ -105,3 +104,32 @@ func (m msgBasedAction) Execute(sim *SimCtx, ctx sdk.Context) (
...
+func totalFrequency(actions []Action) int {
+	totalFrequency := 0
+	for _, action := range actions {
+		totalFrequency += mapFrequencyFromString(action.Frequency())
+	}
+
+	return totalFrequency
+}
+
+type selectActionFn func(r *rand.Rand) Action
```
Comment (p0mvn) [inline simulation/simtypes/action.go:117]: let's define types up top

File: simulation/simtypes/action.go [hunk 3]
```diff
@@ -105,3 +104,32 @@ func (m msgBasedAction) Execute(sim *SimCtx, ctx sdk.Context) (
...
+
+func GetSelectActionFn(actions []Action) selectActionFn {
+	totalOpFrequency := totalFrequency(actions)
+
+	return func(r *rand.Rand) Action {
+		x := r.Intn(totalOpFrequency)
+		// TODO: Change to an accum list approach
```
Comment (p0mvn) [inline simulation/simtypes/action.go:124]: What is that?

Comment (czarcas7ic) [reply simulation/simtypes/action.go:124]: This was a TODO Dev wrote previously and I moved it over, so I am not 100 percent sure

**Review by ValarDragon — APPROVED** [review]

**Review by p0mvn — APPROVED** [review]

**Pull request merged**